src/biokbase/narrative/jobs/jobcomm.py

```diff
@@ -387,14 +471,24 @@ def _get_job_logs(self, req: JobRequest) -> dict:
 
     def _handle_comm_message(self, msg: dict) -> dict:
         """
-        Handles comm messages that come in from the other end of the KBaseJobs channel.
+        Handle incoming messages on the KBaseJobs channel.
+
         Messages get translated into one or more JobRequest objects, which are then
         passed to the right handler, based on the request.
 
         A handler dictionary is created on JobComm creation.
 
         Any unknown request is returned over the channel with message type 'job_error', and a
         JobRequestException is raised.
+
+        :param msg: incoming comm message
+        :type msg: dict
+
+        :raises JobRequestException: if the message type is not recognised
+
+        :return: result of running the appropriate method; generally this is a dictionary
+        of job data indexed by job ID.
+        :rtype: dict
         """
         with exc_to_msg(msg):
```

Comment:
This is so scatter-brained to do ... should be turned into class decorator ideally

Reply:
who on earth might have written such code?! 😉 😁